app/routes.py

```diff
@@ -1,2 +1,312 @@
+                }
+        }, 200
+
+    elif request.method == "DELETE": #deletes from DB
+        db.session.delete(task) # deleting thee instance 
+        db.session.commit()
+        return jsonify({"details": f'Task {task.id} "{task.title}" successfully deleted'}), 200
+        
+#Wave 3 
+@tasks_bp.route("/<task_id>/mark_incomplete", methods = ["PATCH"]) #because the endpoint prefix task is mention on line 18 the endpoint needs to start at/1
+def handle_mark_incomplete(task_id): #because the url is referencing the task_id = 1, the handle function can remain empty but the task variable which holds Task.query.get(1) must have the same task_id value as endpoint
+    
+    task = Task.query.get(task_id) # Task stores the record of id 1, if id 1 doesn't exist what does it store? None 
+    if task is None:
+        return jsonify(None), 404
+    else:
+        task.completed_at = None
+        db.session.commit()
+        
+
+        return {"task":{
+                    "id": task.id,
+                    "title" : task.title,
+                    "description": task.description,
+                    "is_complete": False 
+                    }
+            }, 200
+
+def slack_bot(title):
+    query_path = {"channel": "slack-api-test-channel", "text":title}
+    header = {"Authorization": os.environ.get('BOT')}
+    response = requests.post('https://slack.com/api/chat.postMessage', params = query_path, headers = header)
+    print('Check to see if this works')
+    return response.json()
+
+@tasks_bp.route("/<task_id>/mark_complete", methods = ["PATCH"])
+def handle_mark_complete(task_id):
+
+    task = Task.query.get(task_id) 
+    if task is None:
+        return jsonify(None), 404
+    else:
+        task.completed_at = datetime.now()
+        db.session.commit()
+                    
+        slack_bot(task.title)
+
+        return jsonify( {"task":{
+                    "id": task.id,
+                    "title" : task.title,
+                    "description": task.description,
+                    "is_complete": True 
+                    }
+            }), 200
+        
+
+@goals_bp.route("", methods = ["POST", "GET"])
+def handle_goals():
+
+    if request.method == "GET":
+
+        goals = Goal.query.all()
+
+        each_goal = []
+        
+        for goal in goals:
+
+            each_goal.append({ "id": goal.id,
+                    "title": goal.title}), 200
+
+        return jsonify (each_goal)
+
+    elif request.method == "POST":
+
+        
+        request_body = request.get_json() #request.get_json makes the text readable in json 
+        if "title" not in request_body:
+            return jsonify({"details":"Invalid data"}), 400
+        
+        new_goal = Goal(
+            title= request_body["title"]) #new instance 
+
+        db.session.add(new_goal)
+        db.session.commit()
+        new_goal_response = {"goal":{
+                "id": new_goal.id,
+                "title" : new_goal.title}}
```

Comment:
once you implement the instance method in your goal model you can call that instance method here instead of rewriting the same code